### PR TITLE
Add connect exercise

### PR DESCRIPTION
**Pull request by dalexj**

### CODE REVIEW

File: exercises/connect/example.exs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      {x + 1, y},
+      {x - 1, y},
+      {x + 1, y - 1},
+      {x - 1, y + 1}
+    ]
+    |> Enum.filter(&valid_next_loc(&1, history, board))
+  end
+
+  defp valid_next_loc({x, y}, history, board) do
+    x >= 0 && y >= 0 && x < length(board) && y < byte_size(hd board) && !({x, y} in history)
```
Comment (parkerl) [inline exercises/connect/example.exs]: This is a long chain of conditions. Maybe reformat a bit? 

Also, no tests fail when I remove ` y < byte_size(hd board)`. Missing test?

FYI, I sometimes like to do these like this...

```
    Enum.all? [
      x >= 0,
      y >= 0,
      x < length(board),
      y < byte_size(hd board),
      !({x, y} in history)
    ]
```

It doesn't short circuit but I think it reads kinda nice. Not suggesting a change, just sharing.

**Comment by parkerl** [comment]
@dalexj great job as usual. Just one comment for you to respond to. 

Would you mind if I email you about another matter?

Thanks!

File: exercises/connect/connect_test.exs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ExUnit.configure exclude: :pending, trace: true
+
+defmodule ConnectTest do
+  use ExUnit.Case
+
+  def remove_spaces(rows) do
+    Enum.map(rows, &String.replace(&1, " ", ""))
+  end
+
+  @tag :pending
```
Comment (parkerl) [inline exercises/connect/connect_test.exs]: Oh, we should also have one test where `@tag :pending` is commented out so the user has a place to start.

**Comment by dalexj** [comment]
I've addressed the comments here, good catch on the first `@tag :pending`

feel free to email me - <96dalex@gmail.com>

**Comment by parkerl** [comment]
Great! :tada:

**Pull request merged**